texastribune/donations
# moving static and template folders to /server

#### What's this PR do?
* Moves /static and /templates folders to new /server folder and readjusts references to static accordingly.
* Adds a more fleshed out docker-compose file for local dev
* Adds simplified make commands
* Adds setup.py, so 'server' can be reached

#### Why are we doing this? How does it help us?
This is the last big piece in the effort to clean up and reorganize the donations app. Hopefully this lays a good foundation for further development as we add on to the project.

#### How should this be manually tested?
make sure in your local env-docker file, you have 'SLACK_CHANNEL' set to #tech-test
run `make test` and ensure it runs as expected
run `make` and after a time, you should see the listing of js files that have been built in the console
visit 'localhost/donate' OR 'local.texastribune.org/donate'
ensure the page looks as expected and attempt giving a gift (using the 4242 test card)
ensure you get the "Thank you" page
check the #tech-test channel for a message saying that you pledged x amount

#### How should this change be communicated to end users?
Doesn't need to be.

#### Are there any smells or added technical debt to note?
We'll eventually want to pull all of the routes out of app.py and into their own views.py file (as well as moving some other things around), but that's an easy change at this point and I'd like those updates to be more piecemeal. Plus, they probably take a back seat to actually getting in some stripe updates at this point.

The same goes for vue pieces, which we'll eventually want to pull out of the depths of /static and put in their own folder at the root level, but I'm pushing to wait until our upgrade to vue 3 to handle all of that.

#### What are the relevant tickets?
https://airtable.com/appyo1zuQd8f4hBVx/pagZR0rQ8k70pmwIF?detail=eyJwYWdlSWQiOiJwYWcwTENZcDRHa3VseENuZiIsInJvd0lkIjoicmVjbTJFWkhtblhUUmxCMmIiLCJzaG93Q29tbWVudHMiOmZhbHNlfQ

#### Have you done the following, if applicable:
***(optional: add explanation between parentheses)***


#### TODOs / next steps:
